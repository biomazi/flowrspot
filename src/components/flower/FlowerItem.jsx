import React from 'react'
import styled from 'styled-components';
import Favorite from 'components/ui/Favorite';
import Button from 'components/ui/Button';
import { useSessionContext } from 'hooks/useSessionContext';

const StyledImageContainer = styled.div`
  display:flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: center;
  position: relative;
  width: 280px;
  height: 350px;
  margin: 10px;
  background-size: cover;
  background-position: center;
  background-image: 
    url(${props => props.imgUrl}),
    linear-gradient(180deg, rgba(0, 0, 0, 0.0001) 0%, rgba(0, 0, 0, 0.7) 89.5%);
  p {
      margin: 0;
  }
`

const StyledText = styled.p`
  &&& { margin-bottom: 7px; }
  font-family: Ubuntu;
  font-style: normal;
  font-weight: normal;
  font-size: 20px;
  line-height: 15px;
  text-align: center;
  color: #FFFFFF;
  background-color: transparent;
`;
const StyledLatinText = styled.p`
  font-family: Ubuntu;
  font-style: italic;
  font-weight: normal;
  font-size: 12px;
  line-height: 12px;
  text-align: center;
  color: #FFFFFF;
  mix-blend-mode: normal;
  opacity: 0.7;
  background-color: transparent;
`;

const StyledButton = styled(Button)`
  background: rgba(0, 0, 0, 0.5);
  padding: 9px 15px;
  height: 30px;
  min-width: 103px;
  width: fit-content;
  border-radius: 20px;
  font-family: Ubuntu;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 12px;
  box-shadow: none;
  color: #FFFFFF;
  margin: 20px 0;
`;

const FlowerItem = ({ flower }) => {
  const { authToken } = useSessionContext();
  const isLoggedIn = !!authToken;

  return (
    <StyledImageContainer imgUrl={flower.profile_picture}>
      {isLoggedIn && <Favorite favorite={flower.favorit} />}
      <StyledText>{flower.name}</StyledText>
      <StyledLatinText>{flower.latin_name}</StyledLatinText>
      <StyledButton circular>{flower.sightings} sightings</StyledButton>
    </StyledImageContainer>
  )
};

export default FlowerItem;
