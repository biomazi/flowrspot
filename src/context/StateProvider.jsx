import React, { createContext } from 'react';

export const StateContext = createContext({});

const StateProvider = ({ value, children, ...otherProps }) => {
  return (
    <StateContext.Provider value={{ ...value, ...otherProps }}>
      {children}
    </StateContext.Provider>
  )
}

export default StateProvider;