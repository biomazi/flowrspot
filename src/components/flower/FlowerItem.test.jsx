import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import FlowerItem from './FlowerItem';
import StateProvider from 'context/StateProvider';

const mocks = {
  profile_picture: 'mock.png',
  favorit: true,
  name: 'mock flower',
  latin_name: 'Lorem ipsum dolor sit amet',
  sightings: 5,
}

const renderComponent = contextValue => (
  render(
    <StateProvider value={{ authToken: contextValue }}>
      <FlowerItem flower={mocks} />
    </StateProvider>
  )
);

describe('Test block FlowerItem.jsx', () => {
  test('should render a Flower name', () => {
    renderComponent(null);
    expect(screen.getByText(/mock flower/i)).toBeInTheDocument();
  });

  test('should render number of Flower sightings token', () => {
    renderComponent(null);
    expect(screen.getByText(/5 sightings/i)).toBeInTheDocument();
  });

  test('should render favorite Flower with token present', () => {
    renderComponent('test token');
    expect(screen.getByText(/star.svg/i)).toBeInTheDocument();
  });

  test('should not render favorite Flower without token', () => {
    renderComponent(null);
    expect(screen.queryByText(/star.svg/i)).not.toBeInTheDocument();
  });

  test('should render Flower snapshot', () => {
    const { asFragment } = renderComponent('test token');
    expect(asFragment).toMatchSnapshot();
  });

});