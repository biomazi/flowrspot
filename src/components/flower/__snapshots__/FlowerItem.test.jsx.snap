// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test block FlowerItem.jsx should render Flower snapshot 1`] = `[Function]`;
